### PR TITLE
Pattern Assembler - Delayed pattern previews to use the API cache

### PR DESCRIPTION
**Pull request by miksansegundo**
#### Proposed Changes

* Render the first pattern in each list (headers, sections, and footers) and then the rest after 200 ms. 
This ensures that the cache is created before rendering each group. 

#### Testing Instructions

* **Do not sandbox the API.** Use the API from the production servers.
* Open the Dev tools > Network to see the waterfall of requests. Filter them by `/pattern`. 
* Access `/setup/patternAssembler?siteSlug=[YOUR SITE]`
* Check that all patterns are rendered as expected with the full width of the iframe.
* Check that all the patterns don't load simultaneously but in a waterfall where there are three patterns requested first and then the rest.

Screen Shot 2565-11-01 at 10 07 07

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

**Comment by miksansegundo** [comment]
@fushar I removed the part that delays the sections and footers from this PR so we can merge it soon to solve the 429 error issue. I'll follow up with other PR for loading the headers first.

**Review by fushar — APPROVED** [review]
Tested and the delay works as expected.

**Comment by arthur791004** [comment]
Could we fetch a pattern first and render the patterns later when we get the response? I think it might be safer to ensure the cache is pre-warmed.

200ms delayed might not work when the customer's network connection is not stable 🤔

**Comment by miksansegundo** [comment]
> Could we fetch a pattern first and render the patterns later when we get the response? I think it might be safer to ensure the cache is pre-warmed.

I'll merge this first and let's see if we can still reproduce it. I agree that waiting for the response would be a safer solution.

**Comment by arthur791004** [comment]
Sure!

**Pull request merged**
